src/Planners/include/BipedalLocomotion/Planners/UnicycleTrajectoryPlanner.h

```diff
@@ -0,0 +1,234 @@
+     * | `startAlwaysSameFoot`     |      bool         |       false       |       -         |       Restart with the default foot if still                    |
+     * |     `terminalStep`        |      bool         |       true        |       -         |   Add a terminal step at the end of the horizon                 |
+     * |     `mergePointRatios`    | list of 2 doubles |         -         |   (0.4 0.4)     | The ratios of the DS phase in which it is present a merge point |
+     * | `switchOverSwingRatio`    |     double        |        0.2        |       -         | The ratio between single and double support phases              |
+     * | `lastStepSwitchTime`      |     double        |        0.3        |       -         |       Time duration of double support phase in final step       |
+     * | `isPauseActive`           |      bool         |       true        |       -         |    If true, the planner can pause, instead of make tiny steps.  |
+     * | `comHeight`               |     double        |        0.70       |       -         |    CoM height in double support phase                           |
+     * | `comHeightDelta`          |     double        |        0.01       |       -         |    Delta to add to CoM heinght in Single support phases         |
+     * | `leftZMPDelta`            | list of 2 doubles |         -         |   (0.0  0.0)    | Local ZMP reference: delta wrt center frame of the foot         |
+     * | `rightZMPDelta`           | list of 2 doubles |         -         |   (0.0  0.0)    | Local ZMP reference: delta wrt center frame of the foot         |
+     * | `lastStepDCMOffset`       |     double        |        0.5        |       -         | Last Step DCM Offset. If 0, DCM coincides with stance foot ZMP  |
+     * | `leftContactFrameName`    |     string        |         -         |    "l_sole"     |       Name of the left foot contact frame                       |
+     * | `rightContactFrameName`   |     string        |         -         |    "r_sole"     |       Name of the right foot contact frame                      |
+     *
+    // clang-format on
+
+     * @param handler Pointer to the parameter handler.
+     * @return True in case of success, false otherwise.
+     */
+    bool initialize(std::weak_ptr<const ParametersHandler::IParametersHandler> handler) override;
+
+    const UnicycleTrajectoryPlannerOutput& getOutput() const override;
+
+    bool isOutputValid() const override;
+
+    bool setInput(const UnicycleTrajectoryPlannerInput& input) override;
+
+    bool advance() override;
+
+private:
+    class Impl;
+    std::unique_ptr<Impl> m_pImpl;
+
+    bool setUnicycleControllerFromString(
+    const std::string& unicycleControllerAsString, UnicycleController& unicycleController);
+
+    bool generateFirstTrajectory();
+};
+
+namespace BipedalLocomotion::Planners::Utilities
+{
+bool getContactList(
```

Comment:
Could you comment on the desiderata of this function? It it is only needed on the UnicycleTrajectoryPlanner you may add it there

Reply:
The goal of this function is to populate the ContactList object with the footsteps produced by the `unicycle-footsteps-planner`.

The same function is being deployed by the `UnicycleTrajectoryGenerator` provided by #845. This is why I thought of it as a `Utility`.

Reply:
with commit 94cb121f9116152f790dd01313f15fef4ebd0a71 the `getContactList` has been moved to another file.